wtf-hir/src/compiler.rs

```diff
@@ -15,7 +17,70 @@ use crate::{
 const INTERNAL_PREFIX: &str = "wtfinternal";
 const INTERNAL_SUFFIX: &str = "sdafbvaeiwcoiysxuv";
 
-pub fn compile(ast: ast::Module) -> Module {
+pub fn compile(ast: ast::Module) -> Result<Module, Vec<Error>> {
```

Comment:
please don't use panic catching here but rewrite the functions that might produce errors accordingly.

Also, don't abort on first error but instead collect errors in a struct similar to how the parser is doing it.